flake.nix

```diff
@@ -0,0 +1,337 @@
+                cp ${mbedtlsShim.out}/lib/mbedtls_provider.so standalone/src/main/resources/cz/crcs/ectester/standalone/libs/jni/
+                cp ${wolfcryptjni}/lib/* standalone/src/main/resources/cz/crcs/ectester/standalone/libs/jni/
+                pushd standalone/src/main/resources/cz/crcs/ectester/standalone/libs/jni/
+                make lib_timing.so lib_csignals.so lib_cppsignals.so
+                popd
+              '';
+
+              nativeBuildInputs = [
+                # libresslShim
+                gdb
+                ant
+                jdk17
+                pkg-config
+                global-platform-pro
+                gradle
+                # patched_openssl
+                makeWrapper
+
+                # libraries to test
+                # openssl_3013
+                # boringssl
+                libressl
+                patched_boringssl
+                libtomcrypt
+                libtommath
+                botan2
+                cryptopp
+
+                # libraries' dependencies
+                # cmake
+                # ninja
+                gawk
+                automake
+                go
+                gtest
+                libunwind
+                autoconf
+                libb64
+
+                clang
+                libgcrypt
+                mbedtls
+                nasm
+                libtool
+                perl
+
+                wolfssl
+                nettle
+
+                gmp
+                libgpg-error
+                wget
+                libconfig
+              ];
+
+              buildInputs = [
+                jdk17_headless
+                # libressl
+                # patched_openssl
+              ];
+
+              LD_LIBRARY_PATH = lib.makeLibraryPath [
+                # libresslShim
+                boringsslShim
+
+                libtommath
+                libtomcrypt
+                botan2
+                cryptopp
+                libgcrypt
+                patched_openssl
+                libressl
+                patched_boringssl
+                ninja
+                nettle
+                gmp
+                libgpg-error
+                libconfig
+                wolfcryptjni
+              ];
+
+              BORINGSSL_CFLAGS = "${patched_boringssl.dev.outPath}/include";
+              WOLFCRYPT_LIB_PATH = "${wolfcryptjni}/lib";
+
+              # FIXME more things to copy here
+              installPhase = ''
+                mkdir -p $out
+                cp -r standalone/build $out
+                echo ${opensslVersion} > $out/build/opensslVersion
+              '';
+              
+              postFixup = ''
+                makeWrapper \
+                  ${jdk17_headless}/bin/java $out/bin/${pname} \
+                  --add-flags "-jar $out/build/libs/${pname}.jar" \
+                  --set LD_LIBRARY_PATH ${LD_LIBRARY_PATH}:$LD_LIBRARY_PATH
```

Comment:
Messing with LD_LIBRARY_PATH should not be necessary if statically linked shims are to be used.

Reply:
Yes, this is still a WIP.